app/models/user.rb

```diff
@@ -26,4 +26,15 @@ def favorited(post)
   def voted(post)
     votes.where(post_id: post.id).first
   end
+
+  def self.top_rated
+   self.select('users.*') # Select all attributes of the user
+     .select('COUNT(DISTINCT comments.id) AS comments_count') # Count the comments made by the user
+     .select('COUNT(DISTINCT posts.id) AS posts_count') # Count the posts made by the user
+     .select('COUNT(DISTINCT comments.id) + COUNT(DISTINCT posts.id) AS rank') # Add the comment count to the post count and label the sum as "rank"
+     .joins(:posts) # Ties the posts table to the users table, via the user_id
+     .joins(:comments) # Ties the comments table to the users table, via the user_id
+     .group('users.id') # Instructs the database to group the results so that each user will be returned in a distinct row
+     .order('rank DESC') # Instructs the database to order the results in descending order, by the rank that we created in this query. (rank = comment count + post count)
+   end
```

Comment:
whoa, this is doing a lot. maybe split into a few methods? should never really need comments for people to understand ur code.